src/main/scala/com/fulcrumgenomics/phasing/HapCutToVcf.scala

```diff
@@ -0,0 +1,413 @@
+    new VariantContextBuilder(ctx).genotypes(new GenotypeBuilder(ctx.getGenotype(0)).phased(false).make()).make()
+  }.zipWithIndex.iterator
+  private val hapCutIterator = new HapCutReader(input)
+  private val sampleName = vcfReader.getFileHeader.getSampleNamesInOrder.head
+
+  def hasNext(): Boolean = sourceIterator.hasNext
+
+  /** Returns either the phased variant with added HapCut-specific genotype information (Left), or the original variant (Right). */
+  def next(): Either[VariantContext,VariantContext] = {
+    if (!hasNext()) throw new NoSuchElementException("Calling next() when hasNext() is false")
+    val (sourceContext, sourceOffset) = sourceIterator.next()
+    if (!hapCutIterator.hasNext()) {
+      Right(sourceContext)
+    }
+    else {
+      hapCutIterator.next() match {
+        case None => Right(sourceContext)
+        case Some(hapCutCall) =>
+          if (hapCutCall.offset != sourceOffset+1) throw new IllegalStateException("BUG: calls are out of order")
+          val hapCutContext = HapCutCall.toVariantContext(hapCutCall, sampleName)
+          Left(replaceGenotypes(source = sourceContext, genotype = hapCutContext.getGenotype(0), isPhased = !readBackedPhasingFormat || !hapCutCall.firstInBlock))
+      }
+    }
+  }
+
+  /** Replaces the original genotype with HapCut's genotype and adds any HapCut-specific genotype information. */
+  private def replaceGenotypes(source: VariantContext, genotype: Genotype, isPhased: Boolean): VariantContext = {
+    val builder = new VariantContextBuilder(source)
+    val sourceAlleles = source.getAlleles.toSeq
+    val genotypeAlleles = genotype.getAlleles.map {
+      allele => sourceAlleles.find(a => a.toString == allele.toString) match {
+        case None =>
+          throw new IllegalStateException(s"Could not find allele '$allele' in source alleles: " + sourceAlleles.map{_.toString}.mkString(", "))
+        case Some(a) => a
+      }
+    }
+    val sourceGenotype = source.getGenotype(0)
+    val genotypeBuilder = new GenotypeBuilder(sourceGenotype).alleles(genotypeAlleles).phased(isPhased)
+    genotypeBuilder.attributes(genotype.getExtendedAttributes)
+    builder.genotypes(genotypeBuilder.make()).make()
+  }
+}
+
+/** Haplotype block information produced by HapCut.
+  *
+  * @param phaseSet the phase set identifier for this block (typically the position of the first variant).
+  * @param offset the 1-based variant offset from the first variant given to HapCut.
+  * @param len the total number of variants that this block spans, including unphased variants.
+  * @param phased the total number of variants phased.
+  * @param span the number of bases this block spans.
+  * @param mecScore the minimum error correction (MEC) score.
+  * @param fragments the number of fragments contributing to variants in this block.
+  */
+private class HapCutBlockInfo private(var phaseSet: Int, val offset: Int, val len: Int, val phased: Int, val span: Int, val mecScore: Float, val fragments: Int)
```

Comment:
Looking forward, I see why, but I still don't like it.  Options I think would be nicer:
1) Parse the info in the block line outside the case class, and then only create the block when you have all the information
2) Make it a val and use `copy()` to set it once you have the value (and perhaps use `Option[Int]` instead of `Int` with a special value of `-1`.
3) Screw HapCut's block IDs's and just use the offset as the block ids since those are unique and show up in the block line.